Exclude shared/amdgpu-windows-interop from pre-commit

superm1 opened the pull request:
In order for pre-commit to be useful, everything needs to meet a common baseline.

ScottTodd reviewed (changes requested):
I believe this shared/amdgpu-windows-interop folder still has an external source of truth. We should exclude that whole subtree from any lint checks.

superm1 committed:
Exclude shared/amdgpu-windows-interop from pre-commit rules

These have an external source of truth.
Also drop the non-existent hipblaslt which isn't in rocm-systems.

Signed-off-by: Mario Limonciello (AMD) <superm1@kernel.org>

ScottTodd commented on .pre-commit-config.yaml:
```diff
@@ -4,6 +4,7 @@ exclude: |
     build-.*/
     _build/
```
(can be separate from this PR)

I don't think `build` should need to be added here. Isn't it already in `.gitignore`?

superm1 replied:
Good point.  pre-commit is only supposed to run on things that get committed.  I'll do all the changes in this PR, otherwise they'll be ping-ponging for whichever goes first with so many context lines near by.

superm1 replied:
Oh; actually there is no build* in .gitignore.

ScottTodd commented on .pre-commit-config.yaml:
```diff
@@ -4,6 +4,7 @@ exclude: |
     build-.*/
     _build/
     projects/hipblaslt/library/src/amd_detail/rocblaslt/src/Tensile
```
(can be separate from this PR)

projects/hipblaslt is a path in rocm-libraries, not rocm-systems... so this line is doing nothing here

ScottTodd commented on .pre-commit-config.yaml:
```diff
@@ -4,6 +4,7 @@ exclude: |
     build-.*/
     _build/
     projects/hipblaslt/library/src/amd_detail/rocblaslt/src/Tensile
+    shared/amdgpu-windows-interop
```
Can you add a comment here saying this subfolder has an external source of truth?

You might need the `(?x)` syntax from https://github.com/ROCm/rocm-libraries/blob/2664c167bfbdce7011e53747c7f5094757e56b23/.pre-commit-config.yaml#L5-L17
